app/src/main/java/com/github/sdp/ratemyepfl/model/review/Review.kt

```diff
@@ -1,14 +1,142 @@
 package com.github.sdp.ratemyepfl.model.review
 
+import android.util.Log
 import com.github.sdp.ratemyepfl.serializer.LocalDateSerializer
+import com.google.firebase.firestore.DocumentSnapshot
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
 import java.time.LocalDate
 
 @Serializable
-abstract class Review() {
-    abstract val rating: ReviewRating
-    abstract val title: String
-    abstract val comment: String
+data class Review constructor(
```

Comment:
Good, if I'm not mistaken, the reviewableId is the id for the reviewed item? I think this is the best way to represent the item, even if we lose information in the Review item. Otherwise, we would repeat the Reviewable in every review, and it is memory-consuming.

Reply:
Yep, every reviewable item has at least an id, so we can use to identify.